File: src/tests/test_igamma.cc
/// test_igamma.cc
/// Mac Radigan
/// SPDX-License-Identifier: PDDL-1.0

  #include <complex>
  #include <iostream>
  #include "rad/igamma/igamma.h"

  int main()
  {
    using std::complex;
    using std::cout;
    using std::endl;
    using complex_t = complex<double>;
    using complex128_t = complex<__float128>;
    using namespace rad::igamma;

    {  // double
      complex_t alpha(0.1, 0.1);
      complex_t x(0.1, 0.1);
      complex_t y(0.0, 0.1);
      igamma(alpha, x, y);
      cout << "complex double:" << endl;
      cout << y << " = igamma(" << alpha << ", " << x << ")" << endl << endl;
    }

    {  // binary128
      complex128_t alpha(0.1, 0.1);
      complex128_t x(0.1, 0.1);
      complex128_t y(0.0, 0.1);
      igamma(alpha, x, y);
      cout << "complex binary128:" << endl;
      cout << y << endl
           << " = igamma(" << endl
           << "  " << alpha << endl
           << ", " << x << ")" << endl
           << endl;
    }

    return 0;
  }

/// *EOF*

File: julia/tests/test_igamma.jl
#!/usr/bin/julia -q
### test.jl
### Mac Radigan
### SPDX-License-Identifier: PDDL-1.0

  using Igamma

  a1 = Complex{Float32}(0.1,0.1)
  x1 = Complex{Float32}(0.1,0.1)
  y1 = igamma(a1, x1)
  @printf("(%f,%f) = igamma( a=(%f,%f), x=(%f,%f) )\n", real(y1), imag(y1), real(a1), imag(a1), real(x1), imag(x1))
  println()

# const a2 = big(0.1 + 0.1im)
# const x2 = big(0.1 + 0.1im)
# const y2 = igamma(a2, x2)
# println(y2)
# println()
# @printf(" = igamma(")
# println()
# @printf("  a=")
# println(a1)
# println()
# @printf("  x=")
# println(x1)
# println()
# @printf(")")
# println()

### *EOF*
